ChangeSet
  1.1517 04/03/23 16:36:18 anton@samba.org +3 -0
  use pci_dma_error

  drivers/scsi/ibmvscsi/rpa_vscsi.c
    1.12 04/03/23 16:35:45 anton@samba.org +5 -6
    use pci_dma_error

  drivers/scsi/ibmvscsi/ibmvscsis.c
    1.4 04/03/23 16:35:45 anton@samba.org +5 -4
    use pci_dma_error

  drivers/net/iseries_veth.c
    1.11 04/03/23 16:35:45 anton@samba.org +1 -1
    use pci_dma_error

diff -Nru a/drivers/net/iseries_veth.c b/drivers/net/iseries_veth.c
--- a/drivers/net/iseries_veth.c	Tue Mar 23 09:43:42 2004
+++ b/drivers/net/iseries_veth.c	Tue Mar 23 09:43:42 2004
@@ -455,7 +455,7 @@
 	
 	/* Is it really necessary to check the length and address
 	 * fields of the first entry here? */
-	if (dma_address != NO_TCE) {
+	if (!pci_dma_error(dma_address)) {
 		msg->skb = skb;
 		msg->data.addr[0] = dma_address;
 		msg->data.len[0] = dma_length;
diff -Nru a/drivers/scsi/ibmvscsi/ibmvscsis.c b/drivers/scsi/ibmvscsi/ibmvscsis.c
--- a/drivers/scsi/ibmvscsi/ibmvscsis.c	Tue Mar 23 09:43:42 2004
+++ b/drivers/scsi/ibmvscsi/ibmvscsis.c	Tue Mar 23 09:43:42 2004
@@ -1859,10 +1859,11 @@
 		goto malloc_failed;
 	queue->size = PAGE_SIZE / sizeof(*queue->msgs);
 
-	if((queue->msg_token = vio_map_single(adapter->dma_dev, 
-					      queue->msgs, 
-					      queue->size * sizeof(*queue->msgs), 
-					      PCI_DMA_BIDIRECTIONAL)) == NO_TCE)
+	queue->msg_token = vio_map_single(adapter->dma_dev, queue->msgs,
+					  queue->size * sizeof(*queue->msgs),
+					  PCI_DMA_BIDIRECTIONAL);
+
+	if (pci_dma_error(queue->msg_token))
 		goto map_failed;
 
 	rc = plpar_hcall_norets(H_REG_CRQ, adapter->dma_dev->unit_address, queue->msg_token, PAGE_SIZE);
diff -Nru a/drivers/scsi/ibmvscsi/rpa_vscsi.c b/drivers/scsi/ibmvscsi/rpa_vscsi.c
--- a/drivers/scsi/ibmvscsi/rpa_vscsi.c	Tue Mar 23 09:43:42 2004
+++ b/drivers/scsi/ibmvscsi/rpa_vscsi.c	Tue Mar 23 09:43:42 2004
@@ -165,12 +165,11 @@
 		goto malloc_failed;
 	queue->size = PAGE_SIZE / sizeof(*queue->msgs);
 
-	if ((queue->msg_token = dma_map_single(hostdata->dev,
-					       queue->msgs,
-					       queue->size *
-					       sizeof(*queue->msgs),
-					       PCI_DMA_BIDIRECTIONAL)) ==
-	    NO_TCE)
+	queue->msg_token = dma_map_single(hostdata->dev, queue->msgs,
+					  queue->size * sizeof(*queue->msgs),
+					  PCI_DMA_BIDIRECTIONAL);
+
+	if (pci_dma_error(queue->msg_token))
 		goto map_failed;
 
 	rc = plpar_hcall_norets(H_REG_CRQ,
.........................................................................
# vim: syntax=diff

